mava/wrappers/jumanji.py

```diff
@@ -117,3 +118,74 @@ def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
 
         # Aggregate the list of individual rewards and use a single team_reward.
         return self.aggregate_rewards(timestep, modified_observation)
+    
+
+class ConnectorWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the MA Connector environment."""
+
+    def __init__(self, env: MaConnector):
+        super().__init__(env)
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Connector environment."""
+
+        def create_agents_view(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)   
+            targets = jnp.where((grid % 3 == 0) & (grid != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            my_position = jnp.where(grid == 2, True, False)
+            my_target = jnp.where(grid == 3, True, False)
+            agents_view = jnp.stack((positions, targets, paths, my_position, my_target), -1)
+            return agents_view
+        
+        def create_global_state(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)   
+            targets = jnp.where((grid % 3 == 0) & (grid[0] != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            global_state = jnp.stack((positions, targets, paths), -1)
+            return global_state
+        
+        observation = ObservationGlobalState(
+            global_state = create_agents_view(timestep.observation.grid),
+            agents_view= create_global_state(timestep.observation.grid),
+            action_mask=timestep.observation.action_mask,
+            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
+        ) 
```

Comment:
3 things here:

1. You're always returning `ObservationGlobalState` we should only return (and compute) the global state if `add_global_state` is true, you can see this in other wrappers.
2. You're passing `agents_view` to `global_state` and visa versa
3. You're not vmapping `create_agents_view` so you're only passing out a single observation, there should be n_agents of these, so you need to vmap that over agent_ids to correctly rotate the grid

Reply:
On the last point, create_agents_view works when the observation is batched over the agents, so theres no need to vmap.

Reply:
The rotation is allready done by the Jumanji env. Agent ids are not used.

Reply:
Ah you're right! My bad